src/context/playlistContext.js

```diff
@@ -0,0 +1,159 @@
+import axios from "axios";
+import { createContext, useContext, useReducer } from "react";
+import {
+  SET_ERROR,
+  SET_LOADING,
+  SET_PLAYLISTS,
+  SET_CURRENT_PLAYLIST,
+  SET_VIDEOS,
+} from "reducers/constants";
+import { playlistReducer } from "reducers/reducerFunctions";
+import { initialPlaylistState } from "reducers/states";
+import { useAuth } from "./authContext";
+
+const PlaylistContext = createContext();
+
+const PlaylistProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(playlistReducer, initialPlaylistState);
+  const { playlists, currentPlaylist, loading, error } = state;
+  const {
+    userAuthState: { authToken },
+  } = useAuth();
+
+  const getPlaylists = async () => {
+    dispatch({ type: SET_LOADING });
+    try {
+      const response = await axios.get("/api/user/playlists", {
+        headers: { authorization: authToken },
+      });
+      setTimeout(() => {
```

Comment:
why is there setTimeout here?
is it for maintaining loading state for minimum 1 second?

Reply:
Actually I am making my Api slower so that I can show loader for some extra time